common/djangoapps/util/model_utils.py

```diff
@@ -127,33 +131,35 @@ def truncate_fields(old_value, new_value):
     return {'old': serialized_old_value, 'new': serialized_new_value, 'truncated': truncated_values}
 
 
-def emit_setting_changed_event(user, db_table, setting_name, old_value, new_value):
+def emit_settings_changed_event(user, db_table, changed_fields: Dict[str, Tuple[Any, Any]]):
     """Emits an event for a change in a setting.
 
     Args:
         user (User): the user that this setting is associated with.
         db_table (str): the name of the table that we're modifying.
-        setting_name (str): the name of the setting being changed.
-        old_value (object): the value before the change.
-        new_value (object): the new value being saved.
+        changed_fields: all changed settings, with both their old and new values
 
     Returns:
         None
     """
-    truncated_fields = truncate_fields(old_value, new_value)
+    for (setting_name, old_value, new_value) in changed_fields.items():
```

Comment:
Oooh, that's pretty cool deconstruction.

Reply:
Oops! Pretty, and also wrong. It should be `(setting_name, (old_value, new_value))`.